cypress/e2e/scenario-2.3-embedded-entity.cy.js

```diff
@@ -0,0 +1,249 @@
+/**
+ * Shorthand method to clear the environment being passed.
+ * By default, if no arguments are passed it will target the 'lsm-frontend' environment.
+ *
+ * @param {string} nameEnvironment
+ */
+export const clearEnvironment = (nameEnvironment = "lsm-frontend") => {
+  cy.visit("/console/");
+  cy.get('[aria-label="Environment card"]').contains(nameEnvironment).click();
+  cy.url().then((url) => {
+    const location = new URL(url);
+    const id = location.searchParams.get("env");
+    cy.request("DELETE", `/api/v1/decommission/${id}`);
+  });
+};
+
+/**
+ * based on the environment id, it will recursively check if a compile is pending.
+ * It will continue the recursion as long as the statusCode is equal to 200
+ *
+ * @param {string} id
+ */
+const checkStatusCompile = (id) => {
+  let statusCodeCompile = 200;
+
+  if (statusCodeCompile === 200) {
+    cy.intercept(`/api/v1/notify/${id}`).as("IsCompiling");
+    // the timeout is necessary to avoid errors.
+    // Cypress doesn't support while loops and this was the only workaround to wait till the statuscode is not 200 anymore.
+    // the default timeout in cypress is 5000, but since we have recursion it goes into timeout for the nested awaits because of the recursion.
+    cy.wait("@IsCompiling", { timeout: 10000 }).then((req) => {
+      statusCodeCompile = req.response.statusCode;
+      if (statusCodeCompile === 200) {
+        checkStatusCompile(id);
+      }
+    });
+  }
+};
+
+/**
+ * It will recursively check for instance service status until its "up".
+ *
+ *
+ */
+const checkServiceState = () => {
+  let state = "start";
+
+  if (state !== "up") {
+    // the timeout is necessary to avoid errors.
+    // Cypress doesn't support while loops and this was the only workaround to wait till the statuscode is not 200 anymore.
+    // the default timeout in cypress is 5000, but since we have recursion it goes into timeout for the nested awaits because of the recursion.
+    cy.wait("@GetServiceInventory", { timeout: 10000 }).then((req) => {
+      if (req.response.statusCode !== 200) {
```

Comment:
you might end up in an infinite loop if the server response with a failure.

Reply:
I added exception  for 400 404 and 500, as I think that's the most possible scenarios, but can ask core team if I should include some other status codes